backends/gpu/lib/wrapper/rocsolver_wrapper.cc

```diff
@@ -46,37 +46,57 @@ llvm::Expected<Stream> RocsolverGetStream(rocblas_handle handle) {
 }
 
 llvm::Error RocsolverPotrf(CurrentContext current, rocblas_handle handle,
-                           rocblas_fill fillMode, int n, Pointer<float> A,
-                           int heightA, Pointer<int> devInfo) {
+                           rocblas_datatype dataType, rocblas_fill fillMode, int n,
+                           Pointer<void> A, int heightA, Pointer<int> devInfo) {
   CheckHipContext(current);
-  return TO_ERROR(rocsolver_spotrf(handle, fillMode, n, ToRocm(A), heightA,
-                                   ToRocm(devInfo)));
+  switch (dataType) {
+    case rocblas_datatype_f32_r:
+      return TO_ERROR(rocsolver_spotrf(handle, fillMode, n,
+                      reinterpret_cast<float*>(ToRocm(A)),
+                      heightA, ToRocm(devInfo)));
+    case rocblas_datatype_f32_c:
+      return TO_ERROR(rocsolver_cpotrf(handle, fillMode, n,
+                      reinterpret_cast<rocblas_float_complex*>(ToRocm(A)),
+                      heightA, ToRocm(devInfo)));
+    case rocblas_datatype_f64_r:
+      return TO_ERROR(rocsolver_dpotrf(handle, fillMode, n,
+                      reinterpret_cast<double*>(ToRocm(A)),
+                      heightA, ToRocm(devInfo)));
+    case rocblas_datatype_f64_c:
+      return TO_ERROR(rocsolver_zpotrf(handle, fillMode, n,
+                      reinterpret_cast<rocblas_double_complex*>(ToRocm(A)),
+                      heightA, ToRocm(devInfo)));
+    default:
+      return MakeStringError("Unsupported type: ", dataType);
```

Comment:
Please change to `Printed(dataType)`, so that the error contains the enum name instead of the int value.